File: src/test/java/br/com/jiratorio/base/executionlistener/RestAssuredTestExecutionListener.java
package br.com.jiratorio.base.executionlistener;

import io.restassured.RestAssured;
import org.springframework.test.context.TestContext;
import org.springframework.test.context.support.AbstractTestExecutionListener;

public class RestAssuredTestExecutionListener extends AbstractTestExecutionListener {

    @Override
    public void beforeTestMethod(TestContext testContext) throws Exception {
        String port = testContext.getApplicationContext()
                .getEnvironment().getProperty("local.server.port");
        if (port != null) {
            RestAssured.port = Integer.valueOf(port);
        }
    }

}

File: src/test/java/br/com/jiratorio/base/executionlistener/RestAssuredTestExecutionListener.kt
package br.com.jiratorio.base.executionlistener

import io.restassured.RestAssured
import org.springframework.test.context.TestContext
import org.springframework.test.context.support.AbstractTestExecutionListener

class RestAssuredTestExecutionListener : AbstractTestExecutionListener() {

    override fun beforeTestMethod(testContext: TestContext) {
        RestAssured.port = testContext.applicationContext.environment
                .getProperty("local.server.port")?.toInt() ?: 0
    }

}
